### ISSUE TITLE
poisson/sine does not seem to work for simplex

### ISSUE THREAD

**Issue by nfehn** [exadg/exadg#307]
The application poisson/sine does not work if I simply change `ElementType::Hypercube` to `ElementType::Simplex`. @necioglu could you provide the application in a way that all the other necessary parameters are adjusted if I change to simplex meshes?

**Comment by necioglu** [comment]
I think you are solving this issue in #305. I have seen that you solved this issue by enforcing a set of variables on the user. I personally don't like this design because:

1. The if-else branches with `if(use_simplex)` must be duplicated in each application, if we want to use simplex elements in other applications.
3. The variables are set automatically and could lead to unwanted behavior, without the user knowing. I enjoyed the development procedure by changing parameters one by one, because I had a better overview of the process. When I did set a variable, so that simplex elements worked, I was aware of the possible side effects. For example: I set Element type=Simplex, I see that a distributed Triangulation won't work by a dealii Asssert, I change to fully distributed and know that fully distributed is a new feature of dealii, that I am using my own code for coarse triangulation creation etc. It is also important to mention that the variable combinations, which don't work with simplex elements, are "well asserted" in dealii or in ExaDG.
4. It reduces the flexibility of the "user interface" (which is for me application.h and input.json) of the Poisson/sine problem.
5. Most of the variables which are set in the case of `use_simplex==true` case, could have been set for hypercubes as well. But the if-else branch gives the impression that it is not possible. For example, we could use FullyDistributed in both cases but we chose it only for simplex elements.

This is/was my take on this topic, and the reason why I added Asserts or in some cases made use of the dealii asserts. Of coarse, I am aware that your design has the advantage of changing the element type quickly. For me, the described disadvantages outweigh the advantages.

**Comment by nfehn** [comment]
I understand the concerns in point 4. I am of a different opinion regarding points 1-3. The changes in poisson/sine in #305 add clearly more information to this test case. Without this information, it is too hard to get the simplex case running, as I experienced myself yesterday. Since this is the only test case supporting simplex, everybody (including the most unexperienced user of ExaDG) has to be able to run this test case very easily. The user interface did not become less flexible, because I added the variable use_simplex to a single application. Of course, I am happy about future developments making the code more elegant.
